library/core/src/cmp.rs

```diff
@@ -765,51 +767,187 @@ impl<T: Clone> Clone for Reverse<T> {
 ///
 /// Lexicographical comparison is an operation with the following properties:
 ///  - Two sequences are compared element by element.
-///  - The first mismatching element defines which sequence is lexicographically less or greater than the other.
-///  - If one sequence is a prefix of another, the shorter sequence is lexicographically less than the other.
-///  - If two sequences have equivalent elements and are of the same length, then the sequences are lexicographically equal.
+///  - The first mismatching element defines which sequence is lexicographically less or greater
+///    than the other.
+///  - If one sequence is a prefix of another, the shorter sequence is lexicographically less than
+///    the other.
+///  - If two sequences have equivalent elements and are of the same length, then the sequences are
+///    lexicographically equal.
 ///  - An empty sequence is lexicographically less than any non-empty sequence.
 ///  - Two empty sequences are lexicographically equal.
 ///
 /// ## How can I implement `Ord`?
 ///
-/// `Ord` requires that the type also be [`PartialOrd`] and [`Eq`] (which requires [`PartialEq`]).
+/// `Ord` requires that the type also be [`PartialOrd`] and [`Eq`] which itself requires
+/// [`PartialEq`].
 ///
-/// Then you must define an implementation for [`cmp`]. You may find it useful to use
-/// [`cmp`] on your type's fields.
+/// Non `derive`d implementations of `Ord` require that the [`cmp`] function is implemented
+/// manually. It's a logic error to have [`PartialOrd`] and `Ord` disagree, so it's best practice to
+/// have the logic in `Ord` and implement [`PartialOrd`] as `Some(self.cmp(other))`.
 ///
-/// Here's an example where you want to sort people by height only, disregarding `id`
-/// and `name`:
+/// Conceptually [`PartialOrd`] and `Ord` form a similar relationship to [`PartialEq`] and [`Eq`].
+/// [`PartialEq`] implements and implies an equality relationship between types, and [`Eq`] implies
+/// an additional property on top of the properties implied by [`PartialOrd`], namely reflexivity.
+/// In a similar fashion `Ord` builds on top of [`PartialOrd`] and implies further properties, such
+/// as totality, which means all values must be comparable.
+///
+/// Because of different signatures, `Ord` cannot be a simple marker trait like `Eq`. So it can't be
+/// `derive`d automatically when `PartialOrd` is implemented. The recommended best practice for a
+/// type that manually implements `Ord` is to implement the equality comparison logic in `PartialEq`
+/// and implement the ordering comparison logic in `Ord`. From there one should implement
+/// `PartialOrd` as `Some(self.cmp(other))`.
```

Comment:
This seems redundant with the paragraph two paragraphs ago (lines 784~786)? Written twice?

Same "start with the relevance" ask: what we really want to tell people is they should derive PartialEq, PartialOrd, and Ord, **xor** manually implement all of them, right? Because otherwise they're at risk of being incorrect.

Reply:
I wrote those two paragraphs somewhat disjointed in time, and indeed there is substantial overlap, but the intent is a different one in each. The first tries to communicate the "stacked" nature of the traits, and the second tries to communicate that "no it's not you, it's us, implementing them correctly and idiomatically is rather inconsistent at first glance.". I remember stumbling a couple of times over the fact that of the 3 Options that exist:

A) Implement with logic
B) Derive
C) Implement in terms of other

The correct and idiomatic way to implement the four traits is

- PartialEq: A
- Eq: B
- PartialOrd: C
- Ord: A

Despite sounding very similar *and* having the similar concept of extending the promises made by the partial one, we expect users to follow this at first glance buckwild inconsistency. IMO something like ABAB would much easier to teach and naturally discover, but that comes with its own issues around unwraping and communicating infallibility. What I'm getting at is, I'd like the docs to contain a section that acknowledges the inconsistency.

Reply:
I don't mind repeating things slightly or trying to explain things are inconsistent (though, we also are entitled to pause and say something like "Implementing these traits is *hazard-prone*, you want to do one of these things:"), it just seemed more disjointed than necessary.

Reply:
It's more disjointed than I like as well, do you have an idea how we could improve it?

Reply:
...hm, doesn't deriving Eq always have the "perfect derive" bounds problem?

Reply:
I would say something like:

"Ord requires that the type also be PartialOrd, PartialEq, and Eq.

Because Ord implies a stronger ordering relationship than PartialOrd, and both Ord and PartialOrd must agree, you must choose how to implement Ord **first**. You can choose to derive it, or implement it manually. If you derive it, you should derive all four traits. If you implement it manually, you should manually implement all four traits."

Reply:
I like it, done.